packages/back-end/src/scim/groups/createGroup.ts

```diff
@@ -0,0 +1,55 @@
+import { OrganizationInterface } from "../../../types/organization";
+import { createTeam } from "../../models/TeamModel";
+import { createApiRequestHandler } from "../../util/handler";
+import { GetGroupResponse } from "./getGroup";
+
+interface SCIMMember {
+  value: string; // unclear what the idP expects this to be
+  display: string; // Member's full name or username?
+}
+interface CreateGroupBody {
+  schemas: string[];
+  displayName: string;
+  members: SCIMMember[];
+}
+
+export const createGroup = createApiRequestHandler()(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async (req: any): Promise<GetGroupResponse> => {
+    // console.log("createGroup endpoint was called");
+
+    const requestBody = req.body.toString("utf-8");
+
+    const requestBodyObject = JSON.parse(requestBody) as CreateGroupBody;
+
+    const org = req.organization as OrganizationInterface;
+
+    if (!org) {
+      // Return an error in the shape SCIM is expecting
+    }
+
+    const DEFAULT_TEAM_PERMISSIONS = {
+      role: "collaborator",
+      limitAccessByEnvironment: false,
+      environments: [],
+    };
+
+    const group = await createTeam({
+      name: requestBodyObject.displayName,
+      createdBy: "SCIM",
+      description: "this is a test.",
+      organization: org.id,
+      ...DEFAULT_TEAM_PERMISSIONS,
+    });
+
+    return {
```

Comment:
One pattern that I'm interested in getting your thoughts on is creating a `responseObj` that is, initially a deepClone of the requestBody, and only updating the few parameters we persist on our end.

E.G

```ts
const responseObj = cloneDeep(requestBody);

responseObj.id = user.id;
```

The reasoning behind it is, I'm getting some test failures via the Microsoft SCIM Validator because the response object doesn't include all of the properties included in the request object.

Reply:
I think this makes sense, especially as we consider supporting some more idPs with different requirements. Will implement this and make sure it works as expected